### PR TITLE
TextRendering refactoring

### PR DESCRIPTION
**Pull request by Gillibald**
## What does the pull request do?

-Reworks GlyphRunImpl creation by introducing an abstraction for allocated platform buffers.
-Replaces GlyphTypeface with IGlyphTypeface and removes IGlyphTypefaceImpl

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

### CODE REVIEW

File: src/Avalonia.Base/Platform/IPlatformRenderInterface.cs [hunk 1]
```diff
@@ -171,11 +171,40 @@ public interface IPlatformRenderInterface
...
 
         /// <summary>
-        /// Creates a platform implementation of a glyph run.
+        /// Allocates a platform glyph run buffer.
         /// </summary>
-        /// <param name="glyphRun">The glyph run.</param>
-        /// <returns></returns>
-        IGlyphRunImpl CreateGlyphRun(GlyphRun glyphRun);
+        /// <param name="glyphTypeface">The glyph typeface.</param>
+        ///         /// <param name="fontRenderingEmSize">The font rendering em size.</param>
```
Comment (robloo) [inline src/Avalonia.Base/Platform/IPlatformRenderInterface.cs:177]: XML comments are likely broken several places due to formatting.

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024542-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Base/Platform/IPlatformRenderInterface.cs [hunk 2]
```diff
@@ -171,11 +171,40 @@ public interface IPlatformRenderInterface
...
+        IGlyphRunBuffer AllocateGlyphRun(GlyphTypeface glyphTypeface, float fontRenderingEmSize, int length);
+
+        /// <summary>
+        /// Allocates a horizontal platform glyph run buffer.
+        /// </summary>
+        /// <param name="glyphTypeface">The glyph typeface.</param>
+        /// <param name="fontRenderingEmSize">The font rendering em size.</param>
+        /// <param name="length">The length.</param>
+        /// <returns>An <see cref="IGlyphRunBuffer"/>.</returns>
+        ///  /// <remarks>
```
Comment (robloo) [inline src/Avalonia.Base/Platform/IPlatformRenderInterface.cs:192]: Nit: another

**Review by danwalmsley — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024655-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.Base/Media/FontMetrics.cs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        public int StrikethroughPosition { get; init; }
+
+        /// <summary>
+        ///     Gets a value that indicates the thickness of the underline in design em size.
+        /// </summary>
+        public int StrikethroughThickness { get; init; }
+    }
+}
+
+namespace System.Runtime.CompilerServices
```
Comment (maxkatz6) [inline src/Avalonia.Base/Media/FontMetrics.cs:60]: I think we had a props file for that.

Comment (Gillibald) [reply src/Avalonia.Base/Media/FontMetrics.cs:60]: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Shared/IsExternalInit.cs

Comment (Gillibald) [reply src/Avalonia.Base/Media/FontMetrics.cs:60]: Needs to be included manually

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024699-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024724-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024726-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `11.0.999-cibuild0024784-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Pull request merged**